ios/LayoutReanimation/REAAnimationsManager.m

```diff
@@ -335,14 +329,29 @@ - (BOOL)removeRecursive:(UIView *)view fromContainer:(UIView *)container without
   // start new animations for it, and might as well remove
   // the layout animation config now
   _clearAnimationConfigForTag(view.reactTag);
+
+  // we don't want user interaction on exiting views
+  view.userInteractionEnabled = NO;
+
   return YES;
 }
 
-- (void)removeChildren:(NSArray<UIView *> *)children fromContainer:(UIView *)container
+- (void)reattachAnimatedChildren:(NSArray<id<RCTComponent>> *)children
+                     toContainer:(id<RCTComponent>)container
+                       atIndices:(NSArray<NSNumber *> *)indices
 {
-  for (UIView *removedChild in children) {
-    if (![self removeRecursive:removedChild fromContainer:container withoutAnimation:true]) {
-      [removedChild removeFromSuperview];
+  if (![container isKindOfClass:[UIView class]]) {
+    return;
+  }
+  for (int i = 0; i < children.count; i++) {
+    id<RCTComponent> child = [children objectAtIndex:i];
+    if (![child isKindOfClass:[UIView class]]) {
+      continue;
+    }
+    UIView *childView = (UIView *)child;
+    NSNumber *originalIndex = [indices objectAtIndex:i];
+    if ([self startAnimationsRecursive:childView removingSubviewsWithoutAnimations:true]) {
+      [(UIView *)container insertSubview:childView atIndex:[originalIndex intValue]];
```

Comment:
Is it guaranteed here that indices here are sorted? If not the inserting will put things in a different order or in the worst case we could even try to insert at index that's much greater than the current number of subviews